### PR TITLE
Add polymorphic taxes and shipping to Invoice

### PR DESCRIPTION
**Pull request by 2e3s**
With `manual` and `rebilly` based calculators.

This is an improvement over #177 because a few of certain aspects of invoices obtain a more complex behaviour and the given way works better with OAS and automated tools.
The older fields will be kept for BC since they don't intersect the new ones.

[Preview](https://preview.redoc.ly/rebilly-core/invoice-poly-taxes-shipping/tag/Invoices#operation/PostInvoice)

### CODE REVIEW

File: openapi/components/schemas/Common/CommonInvoice.yaml [hunk 1]
```diff
@@ -55,11 +55,12 @@ properties:
     type: number
     format: double
     readOnly: true
```
Comment (ashkarpetin) [inline openapi/components/schemas/Common/CommonInvoice.yaml:57]: I wonder about this, you removed shippingAmount but left this

File: openapi/components/schemas/Common/CommonInvoice.yaml [hunk 2]
```diff
@@ -55,11 +55,12 @@ properties:
...
-  shippingAmount:
```
Comment (ashkarpetin) [inline openapi/components/schemas/Common/CommonInvoice.yaml:58]: You removed `shippingAmount` but left `taxAmount`. I wonder if it's intentional?

File: openapi/components/schemas/Invoices/TaxTypes/rebilly-taxjar.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+description: Rebilly-based shipping amount calculation with shipping zones.
```
Comment (ashkarpetin) [inline openapi/components/schemas/Invoices/TaxTypes/rebilly-taxjar.yaml:1]: Seems incorrect description

File: openapi/components/schemas/Invoices/InvoiceShipping.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+description: Invoice shipping
+type: object
+discriminator:
+  propertyName: calculator
+  mapping:
+    manual: ./ShippingTypes/manual.yaml
+    rebilly: ./ShippingTypes/rebilly.yaml
+properties:
+  calculator:
```
Comment (slavcodev) [inline openapi/components/schemas/Invoices/InvoiceShipping.yaml:9]: `calculator: manual` does not sound natural, may be we can use another term, e.g. `method` or `type`?

File: openapi/components/schemas/Common/CommonInvoice.yaml (hunk 2 above)
Comment (slavcodev) [reply openapi/components/schemas/Common/CommonInvoice.yaml:58]: Yes. It's migrated to `shipping.amount`. We can add `deprecated` flag to these, but I think it's better to remove them from docs to make sure new integrators will not use them mistakenly.

**Review by slavcodev — DISMISSED** [review]

File: openapi/components/schemas/Common/CommonInvoice.yaml (hunk 2 above)
Comment (2e3s) [reply openapi/components/schemas/Common/CommonInvoice.yaml:58]: Yes, I've removed that too. Deprecation is interesting when generating SDK clients, but we don't do that.

**Review by ashkarpetin — DISMISSED** [review]

**Review by slavcodev — DISMISSED** [review]

File: openapi/components/schemas/Common/CommonInvoice.yaml [hunk 5]
```diff
@@ -44,16 +44,12 @@ properties:
...
-    description: The invoice's shipping amount.
-    type: number
-    format: double
-    readOnly: true
+  shipping:
+    allOf:
+      - $ref: ../Invoices/InvoiceShipping.yaml
+  tax:
+    allOf:
```
Comment (adamaltman) [inline openapi/components/schemas/Common/CommonInvoice.yaml:51]: Why allOf and not just $ref?

Comment (subenson) [reply openapi/components/schemas/Common/CommonInvoice.yaml]: There is no valid reason for this. I changed it

**Review by ashkarpetin — APPROVED** [review]

**Review by justnero — APPROVED** [review]

**Pull request merged**